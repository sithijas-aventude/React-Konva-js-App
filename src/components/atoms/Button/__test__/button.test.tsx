import { Button } from '../index';
import renderer from 'react-test-renderer';

it('renders Button', () => {
  const tree = renderer
    .create(<Button label="Button" icon="ri-admin-line" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders Button - dropDown', () => {
  const tree = renderer
    .create(<Button label="Button text" dropDown />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders Button - Secondary', () => {
  const tree = renderer
    .create(<Button label="Secondary" type="red" icon="ri-admin-line" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
