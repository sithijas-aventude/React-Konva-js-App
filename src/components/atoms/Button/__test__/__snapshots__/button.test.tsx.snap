// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Button - Secondary 1`] = `
<button
  className="btn-base  btn-red "
>
  <i
    className="ri-admin-line"
  />
  <span
    className="text "
  >
    Secondary
  </span>
</button>
`;

exports[`renders Button - dropDown 1`] = `
<button
  className="btn-base   btn-drop-down"
>
  <span
    className=" drop-dwon-text"
  >
    Button text
  </span>
  <i
    className="ri-arrow-down-s-line drop-dwon-icon"
  />
</button>
`;

exports[`renders Button 1`] = `
<button
  className="btn-base   "
>
  <i
    className="ri-admin-line"
  />
  <span
    className="text "
  >
    Button
  </span>
</button>
`;
